Apps/WebClient/src/ClientApp/src/store/modules/patientData/actions.ts

```diff
@@ -0,0 +1,150 @@
+import { ErrorSourceType, ErrorType } from "@/constants/errorType";
+import { HttpError } from "@/models/errors";
+import PatientData, { PatientDataFile } from "@/models/patientData";
+import { LoadStatus } from "@/models/storeOperations";
+import container from "@/plugins/container";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import { ILogger, IPatientDataService } from "@/services/interfaces";
+import { PatientDataActions } from "@/store/modules/patientData/types";
+import {
+    getPatientDataFileState,
+    getPatientDataRecordState,
+} from "@/store/modules/patientData/utils";
+
+export const actions: PatientDataActions = {
+    retrievePatientDataFile(
+        context,
+        params: { hdid: string; fileId: string }
+    ): Promise<PatientDataFile> {
+        const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+        const patientDataService = container.get<IPatientDataService>(
+            SERVICE_IDENTIFIER.PatientDataService
+        );
+
+        return new Promise((resolve, reject) => {
+            const currentState = getPatientDataFileState(
+                context.state,
+                params.fileId
+            );
+            if (
+                currentState.status === LoadStatus.LOADED &&
+                currentState.data !== undefined
+            ) {
+                logger.debug("Patient data file found stored, not querying!");
+                const patientDataFile: PatientDataFile = currentState.data;
+                resolve(patientDataFile);
+            } else {
+                logger.debug("Retrieving patient data file");
+                context.commit("setPatientDataFileRequested", params.fileId);
+                patientDataService
+                    .getFile(params.hdid, params.fileId)
+                    .then((data) => {
+                        if (data === undefined) {
+                            reject(new Error("No patient data was returned"));
+                        } else {
+                            context.commit("setPatientDataFile", {
+                                fileId: params.fileId,
+                                file: data,
+                            });
+                            resolve(data);
+                        }
+                    })
+                    .catch((error: HttpError) => {
```

Comment:
This error would actually be a ResultError. (Applies throughout file.)

All HttpErrors encountered by services should be converted to ResultError by ErrorTranslator on the way out.

As an aside, while it can be useful to have a type annotation here, it doesn't do as much as we'd want - we can't set up different catch blocks for different errors, for example.  All rejections go to the first catch block, regardless of any type mismatches.